File: Programmers/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    
    for (long long h = 1; h * h <= yellow; ++h)
    {
        if (yellow % h == 0)
        {
            long long w = yellow / h;
            
            int totalW = w + 2;
            int totalH = h + 2;
            
            int brownCount = totalW * totalH - yellow;
            
            if (brownCount == brown)
            {
                return { (int)totalW, (int)totalH };
            }
        }
    }
    
    return answer;
}

File: Programmers/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {};
        
        int h = 1;
        
        while ((long) h * h <= yellow) {
            if (yellow % h == 0) {
                int w = yellow / h;
                
                int totalW = w + 2;
                int totalH = h + 2;
                
                int brownCount = totalW * totalH - yellow;

                if (brownCount == brown) {
                    return new int[]{totalW, totalH};
                }
            }
            
            h++;
        }
        
        return answer;
    }
}